From e2938e86fcee9f022a3a6787d9ef0f3ffa74924f Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 13 Feb 2018 19:11:56 +0100
Subject: [PATCH 1/1] fs: fat: avoid useless conversion when calling getcluster

Parameter size of function get_cluster() is of type unsigned long. It makes
no sense to convert actsize to int before passing it to get_cluster as
size.

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 fs/fat/fat.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/fs/fat/fat.c b/fs/fat/fat.c
index dd7888cd6d4..a3c7bf604eb 100644
--- a/fs/fat/fat.c
+++ b/fs/fat/fat.c
@@ -353,7 +353,7 @@ static int get_contents(fsdata *mydata, dir_entry *dentptr, loff_t pos,
 	if (pos) {
 		actsize = min(filesize, (loff_t)bytesperclust);
 		if (get_cluster(mydata, curclust, get_contents_vfatname_block,
-				(int)actsize) != 0) {
+				(unsigned long)actsize) != 0) {
 			printf("Error reading cluster\n");
 			return -1;
 		}
@@ -393,14 +393,16 @@ static int get_contents(fsdata *mydata, dir_entry *dentptr, loff_t pos,
 
 		/* get remaining bytes */
 		actsize = filesize;
-		if (get_cluster(mydata, curclust, buffer, (int)actsize) != 0) {
+		if (get_cluster(mydata, curclust, buffer,
+				(unsigned long)actsize) != 0) {
 			printf("Error reading cluster\n");
 			return -1;
 		}
 		*gotsize += actsize;
 		return 0;
 getit:
-		if (get_cluster(mydata, curclust, buffer, (int)actsize) != 0) {
+		if (get_cluster(mydata, curclust, buffer,
+				(unsigned long)actsize) != 0) {
 			printf("Error reading cluster\n");
 			return -1;
 		}
-- 
2.14.2

